sys/net/gnrc/link_layer/lorawan/gnrc_lorawan.c

```diff
@@ -129,7 +130,7 @@ static void _config_radio(gnrc_lorawan_t *mac, uint32_t channel_freq,
 
     dev->driver->set(dev, NETOPT_IQ_INVERT, &iq_invert, sizeof(iq_invert));
 
-    gnrc_lorawan_set_dr(mac, dr);
+    gnrc_lorawan_set_sf_bw(mac, dr);
```

Comment:
True. But I found the naming of the function to be confusing hence changed the names.

Reply:
> True. But I found the naming of the function to be confusing hence changed the names.

To be honest, I find it more confusing to have a function called `set_sf_bw` that doesn't receive any `sf` and `bw` parameters ^^

Reply:
It's infact setting SF and BW right while DR is param in ?

Reply:
> It's infact setting SF and BW right while DR is param in ?

I think at this point is a taste discussion. But on the technical discussion, the reason why we have a `gnrc_lorawan_set_dr` is because the PHY layer is not well defined (it's scattered between `gnrc_lorawan.c` and `gnrc_lorawan_region.c`.

Ideally, a `gnrc_lorawan_phy.c` file should take care of these stuff (and following the IEEE semantics, that one would only expose the "channel"... or DR in this case). But as it is now, we are assuming the `_region.c` file is implementing the interface for setting the DR settings in the radio.
I'm ok in renaming the function to make it clearer, but merging SF and BW is not a good idea IMO, since this automatically rules out FSK (DR7, no dependency to SF or BW).

Reply:
I have changed it back to `gnrc_lorawan_set_dr`.  Maybe I shall rename it once the PR's related to the ADR stack is merged as I hope to get better clarity after the same.